MaxText/layers/attentions.py

```diff
@@ -277,6 +283,70 @@ def wrap_gpu_flash_attention(query, key, value):
       )
     return wrap_gpu_flash_attention(query, key, value)
 
+  def cudnn_flash_attention(
+    self,
+    query: Array,
+    key: Array,
+    value: Array,
+  ) -> Array:
+    """
+    CUDNN Flash Attention with Transformer Engine.
+    It is an unstable API. In future release, the API can get changed
+    A stable flash attention API will be included soon. Currently,
+    1. It does not support GQA, num_query_heads == num_kv_heads
```

Comment:
nit: remove comments 1 and 2, add asserts.

Reply:
I can add the asserts.. however, the line 318 essentially does the assert within TE.. should I also add them here?

Reply:
Thanks, no need for double assert. Thanks for explaining.